### PR TITLE
feat: SQLAlchemy encrypted field

### PR DESCRIPTION
**Pull request by cofin**
This PR adds an encrypted field for SQLAlchemy

### CODE REVIEW

File: advanced_alchemy/types.py [hunk 1]
```diff
@@ -174,3 +175,42 @@ def process_result_value(self, value: datetime.datetime | None, dialect: Dialect
...
+    def column_expression(self, column: Any) -> Any:
+        return self._handle_encrypted_data(column)
+
+    def _handle_encrypted_data(self, column: Any) -> str:
+        if self.backend == "pgcrypto":
+            return sql_func.pgp_sym_encrypt(column, self.passphrase)  # type: ignore[return-value]
+        if self.backend == "tink":
+            raise NotImplementedError
+        if self.backend == "cryptography":
+            raise NotImplementedError
```
Comment (provinzkraut) [inline advanced_alchemy/types.py:227]: Wouldn't it make more sense to not have these in the first place if they're currently not implemented?

Sounds less confusing to me.

File: advanced_alchemy/types.py [hunk 2]
```diff
@@ -174,3 +175,42 @@ def process_result_value(self, value: datetime.datetime | None, dialect: Dialect
...
+        return None  # type: ignore[unreachable]
```
Comment (provinzkraut) [inline advanced_alchemy/types.py:228]: Could add a
```python
typing.assert_never()
```
to have mypy assert the check is exhaustive.

**Review by provinzkraut — APPROVED** [review]

File: advanced_alchemy/types.py (hunk 1 above)
Comment (cofin) [reply advanced_alchemy/types.py:227]: I expect to have them before this PR is merged.  I was really trying to set up the structure so I could think through the whole thing.

Comment (provinzkraut) [reply advanced_alchemy/types.py:227]: Oh. Sorry for the slightly overzealous comments then :slightly_smiling_face:

**Comment by provinzkraut** [comment]
@cofin is this not good to go?

**Comment by cofin** [comment]
> @cofin is this not good to go?

Yes.  The PGCrypto variant is ready to go.  I was going to finish up the cryptography and tink backends before merging, but maybe that should just get split into a separate PR.

**Comment by provinzkraut** [comment]
> but maybe that should just get split into a separate PR.

Sounds good

**Pull request merged**